### PR TITLE
Add ReferredForPreventativeServices to linelist hts risk categorization

### PR DESCRIPTION
**Pull request by nthusi-codes**
This PR adds the ReferredForPreventativeServices column to the linelist hts risk categorization table. The datapoint is derived from the ReferredServices field in hts client tests

### CODE REVIEW

File: Scripts/REPORTING/load_linelist_hts_risk_categorization_and_test_results.sql [hunk 1]
```diff
@@ -68,13 +69,14 @@ select
...
 		ReasonRefferredForTesting,
-		case
-			when latest_prep_assessment.PatientKey is not null then 1
-			else 0
-		end as ReferredForPreventativeServices
+		ReferredForPreventativeServices
+		-- case
+		-- 	when latest_prep_assessment.PatientKey is not null then 1
+		-- 	else 0
+		-- end as ReferredForPreventativeServices
```
Comment (nobert-mumo) [inline Scripts/REPORTING/load_linelist_hts_risk_categorization_and_test_results.sql:76]: @nthusi-codes @Marymary-dev just confirming: so, the definition here changes to mean that it doesn't necessarily mean if you exist in prep assessments that you were refereed for preventative services?

Comment (nobert-mumo) [reply Scripts/REPORTING/load_linelist_hts_risk_categorization_and_test_results.sql:76]: ah okay, we might need to revise this then: https://github.com/palladiumkenya/dwh-etl/pull/332/files#diff-111c12f3282b9f27653aa6c000d202b1d3bb8757e61be0ef02a55c8905acd5f9R46

File: Scripts/NDWH/HTS_FACT_TABLES/load_FactHTSClientTests.sql [hunk 2]
```diff
@@ -43,6 +43,7 @@ select
     hts_encounter.CoupleDiscordant,
     hts_encounter.Consent,
     hts_encounter.EncounterId,
+    case when hts_encounter.ReferredForServices = 'Yes' and hts_encounter.ReferredServices LIKE '%Pre-Exposure Prophylaxis%' then 1 else 0 end as ReferredForPreventativeServices,
```
Comment (Marymary-dev) [inline Scripts/NDWH/HTS_FACT_TABLES/load_FactHTSClientTests.sql:46]: @nthusi-codes  kindly pick referred services for this column and list the service

**Review by Marymary-dev — APPROVED** [review]
Looks good to me

File: Scripts/REPORTING/load_linelist_hts_risk_categorization_and_test_results.sql [hunk 3]
```diff
@@ -71,7 +72,8 @@ select
 		case
 			when latest_prep_assessment.PatientKey is not null then 1
 			else 0
-		end as ReferredForPreventativeServices
+		end as ReferredForPreventativeServices,
```
Comment (nobert-mumo) [inline Scripts/REPORTING/load_linelist_hts_risk_categorization_and_test_results.sql:75]: @nthusi-codes,  I think `ReferredForPreventativeServices` is redudant here coz the idea was to replace this with the actual `ReferredServices`  which you have already done ?

Comment (nthusi-codes) [reply Scripts/REPORTING/load_linelist_hts_risk_categorization_and_test_results.sql:75]: I  believe the objective was to just the preventive service, not necessarily remove the "ReferredForPreventativeServices" indicator

Comment (nthusi-codes) [reply Scripts/REPORTING/load_linelist_hts_risk_categorization_and_test_results.sql:75]: @Marymary-dev, maybe you can add clarificaton on the task

Comment (nobert-mumo) [reply Scripts/REPORTING/load_linelist_hts_risk_categorization_and_test_results.sql:75]: Sawasawa, if its well documented in the dictionary then that's okay.
LGTM

**Review by nobert-mumo — APPROVED** [review]

**Pull request merged**